src/test/java/org/springframework/data/redis/listener/PubSubResubscribeTests.java

```diff
@@ -147,22 +149,22 @@ void testContainerPatternResubscribe() throws Exception {
 		container.addMessageListener(anotherListener, new PatternTopic(PATTERN));
 
 		// Wait for async subscription tasks to setup
-		Thread.sleep(400);
-
-		// test no messages are sent just to patterns
-		template.convertAndSend(CHANNEL, payload1);
-		template.convertAndSend(ANOTHER_CHANNEL, payload2);
+		await().atMost(Duration.ofMillis(600)).untilAsserted(() -> {
```

Comment:
I utilized Awaitility in use cases like this as it's more handy vs having to resort to `Thread#sleep`. Note that for this particular set of assertions I had to increase the grace period from 400 to 600 ms as the test was flaky, on my machine at least.